src/core/ddsi/src/q_entity.c

```diff
@@ -516,6 +516,69 @@ dds_return_t new_participant_guid (const ddsi_guid_t *ppguid, struct q_globals *
   nn_plist_copy (pp->plist, plist);
   nn_plist_mergein_missing (pp->plist, &gv->default_local_plist_pp, ~(uint64_t)0, ~(uint64_t)0);
 
+#ifdef DDSI_INCLUDE_SECURITY
```

Comment:
The ``new_participant_guid`` function (and not only that one, there are a bunch more with similar names) are already rather unwieldy. Adding 60-odd lines just to copy some properties makes it worse still. Would you mind refactoring this so that the code to fill the property list becomes a separate function?